src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueFalse.cs

```diff
@@ -39,13 +42,25 @@ public void Setup()
 
             _source = values.Skip(Size).Take(Size).ToDictionary(item => item, item => (TValue)(object)item);
             _dictionary = new Dictionary<TKey, TValue>(_source);
+            _hashSet = new HashSet<TKey>(_dictionary.Keys);
             _sortedList = new SortedList<TKey, TValue>(_source);
             _sortedDictionary = new SortedDictionary<TKey, TValue>(_source);
             _concurrentDictionary = new ConcurrentDictionary<TKey, TValue>(_source);
             _immutableDictionary = Immutable.ImmutableDictionary.CreateRange<TKey, TValue>(_source);
             _immutableSortedDictionary = Immutable.ImmutableSortedDictionary.CreateRange<TKey, TValue>(_source);
         }
 
+        [Benchmark]
+        public bool HashSet()
+        {
+            bool result = default;
+            HashSet<TKey> collection = _hashSet;
+            TKey[] notFound = _notFound;
+            for (int i = 0; i < notFound.Length; i++)
+                result ^= collection.TryGetValue(notFound[i], out _);
```

Comment:
The CI fails with:

```log
 error CS1061: 'HashSet<TKey>' does not contain a definition for 'TryGetValue' and no accessible extension method 'TryGetValue' accepting a first argument of type 'HashSet<TKey>'
```

The [method](https://apisof.net/catalog/System.Collections.Generic.HashSet%3CT%3E.TryGetValue(T,T)) got introduced int .NET 4.7.2.

When we were starting the performance repo project, we decided to target .NET 4.6.1 mostly to allow users to open the solution file with old VS. Since using `netcoreapp3.0` requires VS 2019, most our users most probably already have .NET 4.7.2|4.8 SDK installed.

@DrewScoggins @billwert should we update the micro benchmarks project to target `net48` instead of `net461`?

Reply:
@JeffreyZhao could you please use `#if !NETFRAMEWORK` for benchmarks that call `HashSet.TryGetValue`? this should fix the failing CI

```cs
#if !NETFRAMEWORK
        [Benchmark]
        ...
#endif
```

Reply:
For the long term Adam I am fine with moving the targeting to .NET 4.8.

Reply:
Do I still need to add `#if !NETFRAMEWORK` around `TryGetValue` benchmarks?